src/router/ProtectedRoutes.tsx

```diff
@@ -1,10 +1,24 @@
-import { Navigate, Outlet } from 'react-router-dom'
-import { useAuth } from '~components/Auth/useAuth'
+import AccountProtectedRoute from '~src/router/AccountProtectedRoute'
+import OrganizationProtectedRoute from '~src/router/OrganizationProtectedRoute'
+import { RouteObject } from 'react-router-dom'
+import { SuspenseLoader } from '~src/router/SuspenseLoader'
 
-const ProtectedRoutes = () => {
-  const { signerAddress } = useAuth()
-
-  return signerAddress ? <Outlet /> : <Navigate to='/' replace={true} />
-}
+const ProtectedRoutes = ({ children }: { children: RouteObject[] }) => ({
+  element: (
+    <SuspenseLoader>
+      <AccountProtectedRoute />
+    </SuspenseLoader>
+  ),
+  children: [
+    {
+      element: (
+        <SuspenseLoader>
+          <OrganizationProtectedRoute />
+        </SuspenseLoader>
+      ),
+      children,
+    },
+  ],
+})
```

Comment:
Instead of spread-picking the `children`, pass it as a normal argument. This way you'll avoid having to type `{children: []}` every time you're instancing the method.

```suggestion
const ProtectedRoutes = (children: RouteObject[]) => ({
  element: (
    <SuspenseLoader>
      <AccountProtectedRoute />
    </SuspenseLoader>
  ),
  children: [
    {
      element: (
        <SuspenseLoader>
          <OrganizationProtectedRoute />
        </SuspenseLoader>
      ),
      children,
    },
  ],
})
```

Reply:
b18a0c9